src/hardware/input/mouse_config.cpp

```diff
@@ -399,18 +471,39 @@ static void config_init(Section_prop& secprop)
 	        "      Windows 9x under DOSBox. Under Windows 3.x, the driver is not disabled,\n"
 	        "      but the Windows 3.x mouse driver takes over.");
 
-	prop_bool = secprop.Add_bool("dos_mouse_immediate", always, false);
-	assert(prop_bool);
-	prop_bool->Set_help(
-	        "Update mouse movement counters immediately, without waiting for interrupt\n"
-	        "('off' by default). May improve gameplay, especially in fast-paced games\n"
-	        "(arcade, FPS, etc.), as for some games it effectively boosts the mouse\n"
-	        "sampling rate to 1000 Hz, without increasing interrupt overhead.\n"
-	        "Might cause compatibility issues. List of known incompatible games:\n"
-	        "  - Ultima Underworld: The Stygian Abyss\n"
-	        "  - Ultima Underworld II: Labyrinth of Worlds\n"
-	        "Please report it if you find another incompatible game so we can update this\n"
-	        "list.");
+	prop_str = secprop.Add_string("builtin_dos_mouse_driver_model",
+	                              always,
+	                              model_dos_wheel_str);
+	assert(prop_str);
+	prop_str->Set_values(
+	        {model_dos_2button_str, model_dos_3button_str, model_dos_wheel_str});
+	prop_str->Set_help(
+	        "Set the mouse model to be simulated by the built-in DOS mouse driver ('wheel'\n"
+	        "by default).\n"
+	        "  2button:  2 buttons, use for some old games which malfunction if the middle\n"
+	        "            button gets pressed.\n"
+	        "  3button:  3 buttons, only supported by very few games.\n"
+	        "  wheel:    3 buttons + wheel, supports the CuteMouse WheelAPI version 1.0.");
+
+	// TODO: Consider supporting the VBMouse WheelAPI version 2.0 for
+	// dual-wheel support; for now the specification is a draft only and no
+	// software exists which uses this API.
```

Comment:
I don't think we should go out of our way to support new niche DOS developments. The goal of the project is to run DOS games from the 1980-2000 period, and to some extent demos.

Reply:
Well, I'm definitely not planning to add the support until there is some useful software using this API - which, given the current DOS software development (almost non-existing comparing to Amiga, C64, or ZX Spectrum) might never happen.

But I've removed the comment.

Reply:
Yeah, IMO DOS is a bit shit compared to Amiga, C64, Speccy, etc. as retro platforms to do games and demos.

x86 asm is just un-fun compared to 68k and the 6502, and the big advantage of those home computers is that they're essentially fixed hardware. That gives you a stable, common target for developments. There's good reason why writing demos for an Amiga 500 with the 512K expansion became popular again. When you have a moving target like AGA machines with any random turbo card, you might as well write everything in C or just do a 3D demo in C then compile it for Windows, Linux, etc. 😆 

80s fixed platforms challenge creative coders and artists in a way an extensible architecture just can't.